source/game_sa/MenuManager.cpp

```diff
@@ -647,123 +647,180 @@ void CMenuManager::CentreMousePointer() {
-                if (strcmp(key, "SfxVolume") == 0) m_nSfxVolume = atoi(value);
-                else if (strcmp(key, "RadioVolume") == 0) m_nRadioVolume = atoi(value);
-                else if (strcmp(key, "RadioStation") == 0) m_nRadioStation = static_cast<eRadioID>(atoi(value));
-                else if (strcmp(key, "RadioAutoSelect") == 0) m_bRadioAutoSelect = atoi(value) != 0;
-                else if (strcmp(key, "RadioEq") == 0) m_bRadioEq = atoi(value) != 0;
-                else if (strcmp(key, "TracksAutoScan") == 0) m_bTracksAutoScan = atoi(value) != 0;
-                else if (strcmp(key, "RadioMode") == 0) m_nRadioMode = atoi(value);
-                else if (strcmp(key, "UserTrackIndex") == 0) m_nUserTrackIndex = atoi(value);
+            // Parse key=value pairs
+            size_t equalPos = line.find('=');
+            if (equalPos != std::string::npos) {
+                std::string key = line.substr(0, equalPos);
+                std::string value = line.substr(equalPos + 1);
+                
+                // Trim whitespace around key and value
+                key.erase(0, key.find_first_not_of(" \t"));
+                key.erase(key.find_last_not_of(" \t") + 1);
+                value.erase(0, value.find_first_not_of(" \t"));
+                value.erase(value.find_last_not_of(" \t") + 1);
+                
+                // Store in our configuration map
+                configData[currentSection][key] = value;
             }
-            else if (strcmp(section, "Input") == 0) {
-                if (strcmp(key, "MouseAcceleration") == 0) CCamera::m_fMouseAccelHorzntl = (float)atof(value);
-                else if (strcmp(key, "InvertMouseY") == 0) bInvertMouseY = atoi(value) != 0;
-                else if (strcmp(key, "MouseSteering") == 0) CVehicle::m_bEnableMouseSteering = atoi(value) != 0;
-                else if (strcmp(key, "MouseFlying") == 0) CVehicle::m_bEnableMouseFlying = atoi(value) != 0;
-                else if (strcmp(key, "Controller") == 0) m_ControlMethod = atoi(value) != 0;
-                else if (strcmp(key, "InvertPadX1") == 0) m_bInvertPadX1 = atoi(value) != 0;
-                else if (strcmp(key, "InvertPadY1") == 0) m_bInvertPadY1 = atoi(value) != 0;
-                else if (strcmp(key, "InvertPadX2") == 0) m_bInvertPadX2 = atoi(value) != 0;
-                else if (strcmp(key, "InvertPadY2") == 0) m_bInvertPadY2 = atoi(value) != 0;
-                else if (strcmp(key, "SwapPadAxis1") == 0) m_bSwapPadAxis1 = atoi(value) != 0;
-                else if (strcmp(key, "SwapPadAxis2") == 0) m_bSwapPadAxis2 = atoi(value) != 0;
+        }
+        
+        // Helper function to retrieve values with type conversion
+        auto getValue = [&configData](const std::string& section, const std::string& key, auto defaultValue) -> std::decay_t<decltype(defaultValue)> {
+            try:
+                using ValueType = std::decay_t<decltype(defaultValue)>;
```

Comment:
I'll tell you once more, use `CamalCase` for lambdas, and `const auto` for the type.
```suggestion
        const auto GetValue = [&]<typename ValueType>(const std::string& section, const std::string& key, ValueType& defaultValue) -> ValueType {
```